packages/ckeditor5-alignment/src/alignmentediting.js

```diff
@@ -44,24 +45,55 @@ export default class AlignmentEditing extends Plugin {
 		const locale = editor.locale;
 		const schema = editor.model.schema;
 
+		const alignmentOptions = normalizeAlignmentOptions( editor.config.get( 'alignment.options' ) );
+
 		// Filter out unsupported options.
-		const enabledOptions = editor.config.get( 'alignment.options' ).filter( isSupported );
+		const enabledOptions = alignmentOptions.map( option => option.name ).filter( isSupported );
+		const classNameConfig = alignmentOptions.map( option => option.className );
 
 		// Allow alignment attribute on all blocks.
 		schema.extend( '$block', { allowAttributes: 'alignment' } );
 		editor.model.schema.setAttributeProperties( 'alignment', { isFormatting: true } );
 
-		const definition = _buildDefinition( enabledOptions.filter( option => !isDefault( option, locale ) ) );
+		const shouldUseClasses = classNameConfig.filter( className => !!className ).length;
+
+		// There is no need for converting alignment that's the same as current text direction.
+		const options = enabledOptions.filter( option => !isDefault( option, locale ) );
+
+		if ( shouldUseClasses ) {
+			// Upcast and Downcast classes.
+
+			const alignmentClassNames = classNameConfig.reduce( ( classNameMap, className, index ) => {
+				classNameMap[ enabledOptions[ index ] ] = className;
+
+				return classNameMap;
+			}, {} );
```

Comment:
Maybe it would be more readable if we would use `Object.fromEntries()`

Reply:
Done.